### ISSUE TITLE
`google_container_node_pool` creation can't be parallelized

### ISSUE THREAD

**Issue by davidquarles** [terraform-providers/terraform-provider-google#758]
I've been creating multiple node pools attached to a single cluster, created in a separate resource. 
 Every time I try to create more than one additional node pool, the second (and any subsequent) node pool creation fails, because the API call will not succeed against a cluster in any non-`RUNNING` state.  As of now, I'm resorting to `terraform apply -parallelism=1` and explicitly waiting for pending operations in local-exec provisioners to be doubly sure everything will succeed.

My real, eventual goal is creating all the pools directly within the cluster, as the provisioning window will be much smaller, but there are a few remaining obstacles for using that workflow today.  Is this something we can / want to fix directly in the provider?

**Comment by sl1pm4t** [comment]
Hi @davidquarles - I've worked around this issue by using `depends_on` in each node pool, so the cluster + node pools creation happens sequentially, and I don't have to change the `parallelism` flag.

**Comment by ndmckinley** [comment]
Hi there @davidquarles!  I can't seem to repro this in unit tests, where two pools attacked to a single cluster do seem to succeed - can you post the config you're using so I can look at a fix?  :)

**Comment by davidquarles** [comment]
Hi @ndmckinley!  Thanks for taking the time to look at this.  It's been a good many days (and rebases) since i created this issue, and I apologize for not posting the corresponding config back when.  I've done my best to reconstruct what I was doing at the time, which still fails:

```
$ cat main.tf
resource "google_container_cluster" "primary" {
  name = "test"
  zone = "us-west1-a"

  additional_zones = ["us-west1-b"]
  network          = "default"
  subnetwork       = "default"

  master_auth {
    username = ""
    password = ""
  }

  initial_node_count = 1

  lifecycle {
    ignore_changes = ["node_pool"]
  }
}

resource "google_container_node_pool" "np" {
  count              = "3"
  cluster            = "test"
  initial_node_count = "1"
  name               = "test-${count.index}"
  zone               = "us-west1-a"

  autoscaling {
    max_node_count = "1"
    min_node_count = "0"
  }

  node_config {
    machine_type = "g1-small"
  }
}

$ terraform apply # 0.11.0
3 error(s) occurred:

* google_container_node_pool.np[0]: 1 error(s) occurred:

* google_container_node_pool.np.0: Error creating NodePool: googleapi: Error 404: The resource "projects/nordstrom-k8s-test/zones/us-west1-a/clusters/test" was not found., notFound
* google_container_node_pool.np[1]: 1 error(s) occurred:

* google_container_node_pool.np.1: Error creating NodePool: googleapi: Error 404: The resource "projects/nordstrom-k8s-test/zones/us-west1-a/clusters/test" was not found., notFound
* google_container_node_pool.np[2]: 1 error(s) occurred:

* google_container_node_pool.np.2: Error creating NodePool: googleapi: Error 404: The resource "projects/nordstrom-k8s-test/zones/us-west1-a/clusters/test" was not found., notFound
```

I tested with the current terraform version (0.11.2) and the version we were using in November (0.11.0), and it appears to have been fixed upstream between 0.11.0 and 0.11.2.  That is great news!  If anyone happens to know what changed upstream, I'm curious, but we can go ahead and close this issue.
